fvm/storage/derived/table.go

```diff
@@ -198,25 +198,55 @@ func (table *DerivedDataTable[TKey, TVal]) unsafeValidate(
 
 	applicable := table.invalidators.ApplicableInvalidators(
 		txn.toValidateTime)
-	if applicable.ShouldInvalidateEntries() {
-		for key, entry := range txn.writeSet {
-			if applicable.ShouldInvalidateEntry(
+	shouldInvalidateEntries := applicable.ShouldInvalidateEntries()
+
+	for key, entry := range txn.writeSet {
```

Comment:
is this iteration deterministic?

Reply:
It doesn't need to be. The results of this look is either an error is produced or not, and the iteration order does not influence this.

Also since this error is of type `RetryableConflictError` the transaction will be retried after transactions before it have been completed.